### DISCUSSION TITLE
RKE2 with no cloud provider keeps Rancher Manager pending

### DISCUSSION THREAD

**Discussion by framctr** [rancher/rke2#6131] [Q&A] [answered]
Hi, I'm creating RKE2 clusters on OpenStack through Rancher manager and the rancher2 Terraform provider.

I can successfully create clusters when keeping the `disable-cloud-controller` to `false`, but Rancher is keeps `Updating` when `disable-cloud-controller` is set to `true`, because I think some components, like the ingress, remain `Pending` in the cluster.

I need to disable the default RKE2 cloud controller because the openstack cloud controller manager needs to be bind to the same port used by it.

How can I deploy RKE2 without its default cloud controller without keeping Rancher Manager blocked in the `Updating` cluster status?

PS. For security reasons I need to periodically change the openstack credentials and this means that I cannot pass the OpenStack Cloud Controller Manager configuration to the rancher `additiona-manifests` argument.

**Comment by brandond** [comment, marked as answer]
If you disable the built-in CCM, and do not select an alternative one, you must deploy your own replacement CCM, or the nodes will remain tainted uninitialized. The easiest way to do this is to provide in the cluster config a user add-on manifest that specifies a HelmChart resource to install the correct cloud provider for your environment as a bootstrap chart (`bootstrap: true` in the HelmChart spec).

**Reply by brandond** [reply]
> For security reasons I need to periodically change the openstack credentials and this means that I cannot pass the OpenStack Cloud Controller Manager configuration to the rancher additiona-manifests argument.

I think that is a problem you're going to need to solve some other way. The CCM needs to be installed and functional so that the nodes will be untainted, before the rancher components will be able to run.
